From eaa072aa62bb73ba9cc8e322f7621a3a9a72f397 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:43:58 +0800
Subject: [PATCH 05/11] GSI Android 17 compatibility:
 0001-netd-drop-POSIX_SPAWN_CLOEXEC_DEFAULT-to-fix-dnsmasq-crash.patch

Change-Id: Idb6a0c15fcd7b59207c94c2004976ee5610f02e9
---
 server/TetherController.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/TetherController.cpp b/server/TetherController.cpp
index 379c33b..f9a666c 100644
--- a/server/TetherController.cpp
+++ b/server/TetherController.cpp
@@ -291,7 +291,7 @@ int TetherController::startTethering(bool usingLegacyDnsProxy, int num_addrs, ch
         return -res;
     }
     const android::base::ScopeGuard attrGuard = [&] { posix_spawnattr_destroy(&attr); };
-    res = posix_spawnattr_setflags(&attr, POSIX_SPAWN_USEVFORK | POSIX_SPAWN_CLOEXEC_DEFAULT);
+    res = posix_spawnattr_setflags(&attr, POSIX_SPAWN_USEVFORK);
     if (res) {
         ALOGE("posix_spawnattr_setflags failed (%s)", strerror(res));
         return -res;
